meteor/server/api/snapshot.ts

```diff
@@ -711,6 +756,86 @@ export async function restoreFromRundownPlaylistSnapshot(
 	}
 
+	const expectedPackageIdMap = new Map<ExpectedPackageId, ExpectedPackageId>()
+	_.each(snapshot.expectedPackages, (expectedPackage) => {
+		const oldId = expectedPackage._id
+
+		let ownerId:
+			| PieceId
+			| AdLibActionId
+			| RundownBaselineAdLibActionId
+			| BucketAdLibId
+			| BucketAdLibActionId
+			| RundownId
+			| StudioId
+			| undefined = undefined
+
+		switch (expectedPackage.fromPieceType) {
+			case ExpectedPackageDBType.PIECE:
+			case ExpectedPackageDBType.ADLIB_PIECE:
+			case ExpectedPackageDBType.ADLIB_ACTION: {
+				expectedPackage.pieceId = pieceIdMap[unprotectString(expectedPackage.pieceId)]
+				expectedPackage.rundownId = rundownIdMap[unprotectString(expectedPackage.rundownId)]
+				expectedPackage.segmentId = segmentIdMap[unprotectString(expectedPackage.segmentId)]
+				ownerId = expectedPackage.pieceId
+				break
+			}
+			case ExpectedPackageDBType.BASELINE_ADLIB_PIECE:
+			case ExpectedPackageDBType.BASELINE_ADLIB_ACTION: {
+				expectedPackage.pieceId = pieceIdMap[unprotectString(expectedPackage.pieceId)]
+				expectedPackage.rundownId = rundownIdMap[unprotectString(expectedPackage.rundownId)]
+				ownerId = expectedPackage.pieceId
+				break
+			}
+			case ExpectedPackageDBType.RUNDOWN_BASELINE_OBJECTS: {
+				expectedPackage.rundownId = rundownIdMap[unprotectString(expectedPackage.rundownId)]
+				ownerId = expectedPackage.rundownId
+				break
+			}
+			case ExpectedPackageDBType.BUCKET_ADLIB:
+			case ExpectedPackageDBType.BUCKET_ADLIB_ACTION: {
+				expectedPackage.pieceId = pieceIdMap[unprotectString(expectedPackage.pieceId)]
+				ownerId = expectedPackage.pieceId
+				break
+			}
+			case ExpectedPackageDBType.STUDIO_BASELINE_OBJECTS:
+				ownerId = expectedPackage.studioId
+				break
+
+			default:
+				assertNever(expectedPackage)
+		}
+		if (ownerId && expectedPackage.blueprintPackageId) {
+			expectedPackage._id = getExpectedPackageId(ownerId, expectedPackage.blueprintPackageId)
+		}
+
+		expectedPackageIdMap.set(oldId, expectedPackage._id)
+	})
+	_.each(snapshot.expectedPackageWorkStatuses, (expectedPackageWorkStatus) => {
+		_.each(expectedPackageWorkStatus.fromPackages, (fromPackage) => {
+			fromPackage.id = expectedPackageIdMap.get(fromPackage.id) || protectString('N/A')
+		})
+		expectedPackageWorkStatus._id = getRandomId()
+	})
+	_.each(snapshot.packageContainerPackageStatuses, (packageContainerPackageStatus) => {
+		packageContainerPackageStatus.packageId =
+			expectedPackageIdMap.get(packageContainerPackageStatus.packageId) || protectString('N/A')
+
+		packageContainerPackageStatus._id = getPackageContainerPackageId(
+			packageContainerPackageStatus.studioId,
+			packageContainerPackageStatus.containerId,
+			packageContainerPackageStatus.packageId
+		)
+	})
+	snapshot.packageContainerPackageStatuses = snapshot.packageContainerPackageStatuses.filter(
+		(p) => p.packageId !== protectString('N/A')
+	)
+
+	_.each(snapshot.packageInfos, (packageInfo) => {
```

Comment:
This would be nicer to do as a single loop that builds a new array, instead of using a temporary magic value. Same for all the loops above.
eg
```
let restorePackageInfos = []
_.each(snapshot.packageInfos, (packageInfo) => {
  const newPackageId = expectedPackageIdMap.get(packageInfo.packageId)
  if (newPackageId) {
    packageInfo.packageId = newPackageId
    restorePackageInfos.push(packageInfo)
  }
})
```